Make Profile/Profile Views retain the same order of columns as original DataFrame

FelipeAdachi opened the pull request:
When logging a dataframe, order of the columns was not retained in profile/profile_view.

The problem is that `frozenset` doesn't retain order.

- Changed `schema's` `get_col_names` to return a tuple instead of frozenset (to ensure order and immutability)
- Replaced `difference()` (that only works with sets) with logic that works with tuples
- Added test to assert that order is retained

## Description

This repository needs <!-- Add 1-2 sentences explaining the purpose of this PR. -->

The commits in this pull request will <!-- Summarize PR. -->

## Changes

- Imperative commit or change title (commit ID) <!-- Add commit ID and [GitHub will autolink](https://help.github.com/en/github/writing-on-github/autolinked-references-and-urls). -->
  - Add some bullet points to explain the change
- Next commit or change (commit ID)

## Related

Relates to organization/repo#number

<!-- Reference related commits, issues and pull requests. Type `#` and select from the list. -->

Closes organization/repo#number

<!-- List issues for GitHub to automatically close after merge to default branch. Type `#` and select from the list. See the [GitHub docs on linking PRs to issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) for more. -->

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

FelipeAdachi committed:
- changed schema.py get_col_names to return a tuple
- changed difference operation to work with tuples instead of sets
- added tests to ensure that order is retained when logging

TheMellyBee commented:
One thing I'd recommend is to go with an OrderedDict from collections (which is almost the same as the default dict 3.7+ but good to use from collections for capability). This will maintain the O(1) look up we had with the set which will make your list comprehension more efficient while maintaining the order.  

Now that does remove the immutability which I'm glad you pointed out. In this case since we are getting the names of something are we super worried about getting the immutability? If so then Tuples would be best way to go. 

Personally, it's a pretty common practice to return a list of keys which is mutable. I personally think the api would be in line with others. What do you think?

FelipeAdachi committed:
- changed get_col_names to return dict_keys
- added KeyView from `collections` to annotate dict_keys
- difference operation being done with dict_keys and dict instead of tuples

TheMellyBee commented on python/whylogs/core/dataset_profile.py:
```diff
@@ -108,7 +108,9 @@ def _do_track(
         if dirty:
-            new_cols = self._schema.get_col_names().difference(self._columns.keys())
+            schema_col_keys = self._schema.get_col_names()
+            col_keys = tuple(self._columns.keys())
```
Actually instead of changing the types, you can change up lines 112 and 113 to be able to use a hash more optimally for the difference using by using this:

`new_cols = (col for col in schema_col_keys if col not in self._columns)`

FelipeAdachi replied:
yes, that's much better!
Thanks, Melly!
changed in the last commit!

FelipeAdachi committed:
- don't convert dict into tuple for `col_keys` for efficiency

TheMellyBee reviewed (approved):
Looks good! :D
